File: mvvm/JetLiYu/app/src/main/java/pri/weiqiang/jet/liyujapanese/data/dao/LessonDao.java
package pri.weiqiang.jet.liyujapanese.data.dao;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Query;
import pri.weiqiang.jet.liyujapanese.data.bean.Lesson;

@Dao
public interface LessonDao {
    @Query("SELECT * FROM lessons WHERE book like :book")
    LiveData<List<Lesson>> getLessonByBook(String book);

    @Query("SELECT * FROM lessons")
    LiveData<List<Lesson>> getAll();
}

File: mvvm/KtJetLiYu/app/src/main/java/pri/weiqiang/jet/kt/liyujapanese/data/dao/LessonDao.kt
package pri.weiqiang.jet.kt.liyujapanese.data.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Query
import pri.weiqiang.jet.kt.liyujapanese.data.bean.Lesson

@Dao
interface LessonDao {
    @Query("SELECT * FROM lessons WHERE book like :book")
    fun getLessonByBook(book: String): LiveData<List<Lesson>>

    @Query("SELECT * FROM lessons")
    fun getAll(): LiveData<List<Lesson>>
}